### PR TITLE
Distcc & ccache support

### PR DESCRIPTION
**Pull request by Arthur2e5**
Written according to the manpage.

Tell me about #45 and I will implement those.

### CODE REVIEW

File: proc/12-parallel.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+abinfo "Parallel build ENABLED"
+# Advanced building. Need some code cleaning.
+if ! bool $NOPARALLEL && boolopt NO_DISTCC && _which distcc &>/dev/null; then
+	ab_distcc=1
+	if boolopt NO_DISTCCPUMP && _which pump &>/dev/null; then
+		ab_distccpump=1
+		MAKE_PREFIX="pump"
+	fi
+	abinfo "Going DISTCC!"
+	ABTHREADS=${DISTCC_THREADS:-$((ABTHREADS*2))}
```
Comment (Arthur2e5) [inline proc/12-parallel.sh]: A weird guess.

File: proc/12-parallel.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Still, let's use ABMK. MAKEFLAGS should be in defines.
+: ${ABTHREADS=$(( $(nproc) * 2 + 1))}
```
Comment (Arthur2e5) [inline proc/12-parallel.sh]: That's already lots of threads.. *2.

Comment (MingcongBai) [inline proc/12-parallel.sh]: @Arthur2e5 times-two should be removed.

**Comment by Arthur2e5** [comment]
The deadline is near(

**Pull request merged**